### PR TITLE
zh-Hans: Translation for basic_syntax/references.md

### PR DESCRIPTION
**Pull request by anlunx**
for issue https://github.com/google/comprehensive-rust/issues/324

### CODE REVIEW

File: po/zh-Hans.po [hunk 1]
```diff
@@ -2536,11 +2536,11 @@ msgstr ""
 
 #: src/basic-syntax/references.md:1
 msgid "# References"
-msgstr ""
+msgstr "# 引用"
 
 #: src/basic-syntax/references.md:3
 msgid "Like C++, Rust has references:"
-msgstr ""
+msgstr "如同C++一样，Rust也提供了引用类型。"
```
Comment (wnghl) [inline po/zh-Hans.po:2543]:
```suggestion
msgstr "如同 C++ 一样，Rust 也提供了引用类型。"
```

File: po/zh-Hans.po [hunk 2]
```diff
@@ -2567,6 +2567,9 @@ msgid ""
 "* References that are declared as `mut` can be bound to different values "
 "over their lifetime."
 msgstr ""
+"* 就像C与C++中的指针一样，对引用`ref_x`进行赋值时，我们会对其解引用。\n"
+"* Rust有时会进行自动解引用。比如调用方法 `ref_x.count_ones()` 时，ref_x会被解引用。\n"
+"* 如果引用值在被声明为`mut`（可变引用），那么这个引用值可以在它的生命周期内被绑定为不同的值。"
```
Comment (wnghl) [inline po/zh-Hans.po:2572]:
```suggestion
"* 就像 C 与 C++ 中的指针一样，对引用 `ref_x` 进行赋值时，我们必须对其解引用。\n"
"* Rust 有时会进行自动解引用。比如调用方法 `ref_x.count_ones()` 时，ref_x 会被解引用。\n"
"* 如果引用值被声明为 `mut`（可变引用），那么这个引用值可以在它的生命周期内被绑定为不同的值。"
```

File: po/zh-Hans.po [hunk 3]
```diff
@@ -2577,6 +2580,8 @@ msgid ""
 "  different values, while the second represents a reference to a mutable "
 "value."
 msgstr ""
+"* 注意`let mut ref_x: &i32` 与 `let ref_x: &mut i32` "
+"之间的区别。第一条语句声明了一个可变引用，所以我们可以修改这个引用所绑定的值；第二条语句声明了一个指向可变变量的引用。"
```
Comment (wnghl) [inline po/zh-Hans.po:2584]:
```suggestion
"* 注意 `let mut ref_x: &i32` 与 `let ref_x: &mut i32` "
"之间的区别。第一条语句声明了一个可变引用，所以我们可以修改这个引用所绑定的值；第二条语句声明了一个指向可变变量的引用。"
```

**Review by wnghl — COMMENTED** [review]
Great translationsi🎉🎉🎉! But maybe we could add space between Chinese and English words for better display effect when compiled to markdown file

**Comment by anlunx** [comment]
Thanks for the suggestion! Fixed the format.

**Review by wnghl — APPROVED** [review]

**Pull request merged**